cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/SsaInternalsCommon.qll

```diff
@@ -385,13 +391,50 @@ predicate isModifiableByCall(ArgumentOperand operand) {
       else any()
     else
       // An argument is modifiable if it's a non-const pointer or reference type.
-      exists(Type t, boolean isGLValue | type.hasType(t, isGLValue) |
-        // If t is a glvalue it means that t is always a pointer-like type.
-        isGLValue = true
```

Comment:
I'm clearly missing something. Why is this no longer needed?

Reply:
Oh, right. Good question. I couldn't actually find a case where a _non-qualifier_ argument was a `glvalue` (and I'm not sure this is actually possible?). So I scrapped this case. And as DCA shows, it doesn't seem to contribute to any additional flow.

Reply:
Looking at some IR examples, we always seem to do a `Convert` to a pointer if the operand is a glval.